contracts/Marketplace.sol

```diff
@@ -82,6 +81,38 @@ contract Marketplace is Collateral, Proofs {
     }
   }
 
+  function _freeSlot(
+    bytes32 slotId
+  ) internal marketplaceInvariant {
+    Slot storage slot = _slot(slotId);
+    bytes32 requestId = slot.requestId;
+    RequestContext storage context = requestContexts[requestId];
+    require(context.state == RequestState.Started, "Invalid state");
```

Comment:
If a hosts misses many proofs before the contract is started, then this line will make `Storage.markProofAsMissing()` fail.

We should probably allow the `New` state here as well. When a host misses many proofs in the `New` state we should probably free the slot, but not move to the `Failed` state when not enough slots are filled.